### PR TITLE
XW-821 | don't set properties in didInsertElement

### PR DESCRIPTION
**Pull request by hakubo**
as this is bad for performance
and is being deprecated

### CODE REVIEW

File: front/common/modules/VideoLoader.js [hunk 1]
```diff
@@ -47,6 +46,8 @@ export default class VideoLoader {
 		this.player = VideoLoader.createPlayer(playerClass, provider, params);
 		this.player.setupPlayer();
 		this.player.onResize();
+
+		return this.player;
```
Comment (vforge) [inline front/common/modules/VideoLoader.js]: Please update JSDoc

File: front/main/app/components/lightbox-video.js [hunk 2]
```diff
@@ -40,15 +46,11 @@ export default Ember.Component.extend(
 		 * @returns {void}
 		 */
 		initVideoPlayer() {
-			const videoLoader = new VideoLoader(this.get('model.embed')),
-				selector = Ember.get(videoLoader, 'player.containerSelector');
+			const videoLoader = this.get('videoLoader'),
+				player = videoLoader.loadPlayerClass();
```
Comment (vforge) [inline front/main/app/components/lightbox-video.js]: As only `player.containerSelector` is needed why you need whole `player` object?

Comment (hakubo) [inline front/main/app/components/lightbox-video.js]: did not want to change too much at this point
and returning only containerSelector from loadPlayerClass seemed odd to me

File: front/main/app/components/lightbox-video.js [hunk 3]
```diff
@@ -21,16 +20,23 @@ export default Ember.Component.extend(
 		 * As soon as the videoLoader will be set, the property will be changed.
```
Comment (rogatty) [inline front/main/app/components/lightbox-video.js]: Is this description still valid? `this.videoLoader` shouldn't be empty anymore as it's a computed property, right?

**Comment by rogatty** [comment]
Two docs to update and it'll LGTM

File: front/common/modules/VideoLoader.js (hunk 1 above)
Comment (bognix) [inline front/common/modules/VideoLoader.js]: :+1:

**Comment by bognix** [comment]
Minor issues, otherwise LGTM

**Comment by tomasznapieralski** [comment]
LGTM

File: front/main/app/components/lightbox-video.js [hunk 4]
```diff
@@ -40,15 +49,34 @@ export default Ember.Component.extend(
...
-			this.$(selector).click(() => {
-				return false;
-			});
+			this.$(this.wrapperClass).on(`click.${this.id}`, () => false);
+		},
 
-			this.set('videoLoader', videoLoader);
+		/**
+		 * Since we don't use Ember to inject video HTML
+		 * we have to manage it manually
```
Comment (rogatty) [inline front/main/app/components/lightbox-video.js]: Let's expand this description so everybody understands why we need to do this.

File: front/common/modules/VideoLoader.js [hunk 5]
```diff
@@ -69,7 +68,7 @@ export default class VideoLoader {
 	 * @param {string} playerClass
 	 * @param {string} provider
 	 * @param {Object} params
-	 * @returns {*}
+	 * @returns {BasePlayer|OoyalaPlayer|YouTubePlayer}
```
Comment (vforge) [inline front/common/modules/VideoLoader.js]: :+1:

File: front/main/app/components/lightbox-video.js [hunk 6]
```diff
@@ -40,15 +49,39 @@ export default Ember.Component.extend(
...
+		},
 
-			this.set('videoLoader', videoLoader);
+		/**
+		 * Since we don't use Ember to inject video HTML
+		 *
+		 * Video code does interact with the DOM and brakes binding for Ember templates
+		 * therefore whenever video changes in lightbox
+		 * Ember would not know what to update
+		 *
```
Comment (vforge) [inline front/main/app/components/lightbox-video.js]: Why this newline here?

Comment (hakubo) [inline front/main/app/components/lightbox-video.js]: I like new lines :)

**Pull request merged**